### PR TITLE
Add RTC HAL page to PG

### PR DESCRIPTION
**Pull request by bulislaw**
Initial draft of RTC HAL page for PG

CC: @c1728p9 @0xc0170

### CODE REVIEW

**Comment by AnotherButler** [comment]
@bulislaw I've copy edited this document. Please review it to make sure I didn't accidentally change content. Also, please be sure to add missing links.

**Comment by bulislaw** [comment]
Thanks, looks great.

File: docs/contributing/rtc.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+### Implementing the RTC API
+
+RTC HAL API is in [hal/rtc_api.h](/docs/v5.4/mbed-os-api-doxy/rtc__api_8h_source.html). You need to implement the following functions to support RTC:
+
+To enable sleep support in Mbed OS, you need to add the `RTC` label in the `device_has` option of the target's section in the `targets.json` file.
+
+### Testing
+
+The [RTC HAL API test suite]() validates:
```
Comment (AnotherButler) [inline docs/contributing/rtc.md]: Query: Where should this link?

Comment (AnotherButler) [reply docs/contributing/rtc.md]: @0xc0170 Can you tell me where this should go?

Comment (AnotherButler) [reply docs/contributing/rtc.md]: @c1728p9 Do you know where this link should go?

Comment (c1728p9) [reply docs/contributing/rtc.md]: This is still under review here:
https://github.com/ARMmbed/mbed-os/pull/5008

**Pull request merged**